templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -175,6 +175,11 @@ contents:
         echo "OVS successfully configured"
         ip a show br-ex
         configure_driver_options ${iface}
+        if [ -e /etc/keepalived/keepalived.conf ]; then
+          # Stop keepalived after switching the default interface. 
```

Comment:
Would it be better to make keepalived be `After: network-online.target` ?

Reply:
keepalived runs in static pod. So it is `After: network-online.target`. However, the `keepalived-monitor` depends on `kube-apiserver` to generate the correct `keepalived.conf`.  We cannot control the order of the starting of static pods. So when keepalived is up before kube-apiserver without a up-to-date config, it will mistakenly set the VIP to the old interface and break the network connectivity. Then kube-apiserver cannot come up as expected. We end up in a  dead lock.

The logic here is just a workaround for this case, so that keepalived can not start before kube-apiserver. Ideally, keepalived side shall be able to handle this situation properly after fixing https://bugzilla.redhat.com/show_bug.cgi?id=1873955. But before that is available, this workaround can work.